src/views/components/item.jsx

```diff
@@ -1,16 +1,57 @@
 import React from 'react';
-
+import ItemStore from '../../store/item-store.js';
+import { Link } from 'react-router';
+import { searchItem } from '../../lib/api-client.js';
 export default class Item extends React.Component {
+    constructor(props) {
+        super(props);
+        if (props.item === undefined)
+            this.state = {item: null};
+        else
+            this.state = {item: props.item};
+    }
 
-    render() {
-        if (this.props.item === null) {
-            return <i>No item found</i>;
+    componentDidMount() {
+        if (this.state.item === null) {
+            searchItem(this.props.params.address).then(() => {
+                this.setState({
+                    item: ItemStore.getItem()
+                });
+            });
```

Comment:
... Does it work?

I'm to afraid to try running it, but I don't know how `.then` would work here. `searchItem` in `api-client.js` doesn't return a promise. It doesn't return anything.